WordPress/Sniffs/PHP/StrictInArraySniff.php

```diff
@@ -51,13 +52,26 @@ class StrictInArraySniff extends AbstractFunctionParameterSniff {
 	 *
 	 * @since 0.10.0
 	 * @since 0.11.0 Renamed from $array_functions to $target_functions.
+	 * @since 3.0.0  The format of the array value has changed from boolean to array.
 	 *
-	 * @var array <string function_name> => <bool always needed ?>
+	 * @var array <string function_name> => <array>
```

Comment:
Needs a separate issue / PR, but I wonder if we could document the array / support IDE auto-completion / static analysis a little more:

```suggestion
	 * @var array<string, array{param_position: int, param_name: string, always_needed: bool}>
```

Reply:
Is this psalm style for docblock annotations? In phpstan that would be something along these lines

```php
 * @var []array<string, mixed>
```

But this is something that should be addressed here: https://core.trac.wordpress.org/ticket/56070 (there is a call to make a proposal for a consistent doc standard in WordPress).

Reply:
I'm not aware that is a supported format for any form of static analysis.

I think this should be addressed separately when the docs standards are being improved (any PR for this would need to include making sure the docs for the sniffs comply).

I mean, we could use the WP array format here, but let's be honest, that's a gotspe and not recognized nor supported anywhere outside of WP anyway.

Reply:
P.S.: I've fixed the space ;-)

Reply:
@dingo-d That ticket is only about the tag order, not about the param type formatting.

Reply:
When the keys are known, you can use Array Shapes:

- https://phpstan.org/writing-php-code/phpdoc-types#array-shapes
- https://blog.jetbrains.com/phpstorm/2021/07/phpstorm-2021-2-release/#Array_shapes_PHPDoc_syntax_support
- https://psalm.dev/docs/annotating_code/type_syntax/array_types/#array-shapes

It's a step further than the regular `array<string, array>` format, but relatively well supported amongst tooling.